processor/transformprocessor/internal/common/processor.go

```diff
@@ -0,0 +1,358 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/transformprocessor/internal/common"
+import (
+	"fmt"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	"go.uber.org/multierr"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/contexts/ottldatapoints"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/contexts/ottllogs"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/contexts/ottlresource"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/contexts/ottlscope"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/contexts/ottltraces"
+)
+
+type Context interface {
```

Comment:
@kentquirk here is a "interface wrapper" we discussed.  It allows us to have the dummy `IsContext` function instead of defining the empty `Process[Signal]` functions.  When being used, it'll be cast to the signal specific version, such as 

```go
for _, contexts := range p.contexts {
  contexts.(common.TracesContext).ProcessTraces(td)
}
```

Reply:
Looking at it now, I personally kind of like it this way (although I might make it just `func IsContext()` (no return value) and then it just becomes a one-liner: `func (r *ResourceStatements) IsContext() {}`. It is also easier to mock in testing this way.

But I don't feel *too* strongly about it, so if you do, I'm happy to leave it.

Reply:
We'd actually need to check the cast right? Like:

```go
for _, context := range p.contexts { 
  logContext, ok := context.(common.TracesContext)
  if !ok {
    ...handle error...
  }
  logContext.ProcessTraces(td) 
}
```